### PR TITLE
Avoid throws Exception where possible - Phase III

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

File: spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/metadata/ZookeeperMetadataStore.java [hunk 1]
```diff
@@ -120,14 +122,15 @@ public String putIfAbsent(String key, String value) {
...
+			catch (@SuppressWarnings(UNUSED) KeeperException.NodeExistsException e) {
 				// so the data actually exists, we can read it
 				try {
 					byte[] bytes = this.client.getData().forPath(getPath(key));
 					return IntegrationUtils.bytesToString(bytes, this.encoding);
 				}
 				catch (Exception exceptionDuringGet) {
-					throw new ZookeeperMetadataStoreException("Exception while reading node with key '" + key + "':", e);
+					throw new ZookeeperMetadataStoreException("Exception while reading node with key '" + key + "':",
+							exceptionDuringGet);
```
Comment (garyrussell) [inline spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/metadata/ZookeeperMetadataStore.java]: This is a bug, needs backporting (wrong exception).

File: spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/lock/ZookeeperLockRegistry.java [hunk 2]
```diff
@@ -300,10 +299,16 @@ public Boolean call() throws Exception {
...
+			catch (@SuppressWarnings("unused") TimeoutException e) {
+				if (future != null) {
+					future.cancel(true);
+				}
 				return false;
 			}
+			catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				throw e;
+			}
```
Comment (garyrussell) [inline spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/lock/ZookeeperLockRegistry.java]: These two are bugs (need backporting).

**Review by garyrussell — COMMENTED** [review]
This also fixes a couple of bugs in zookeeper that will need back-porting.

File: spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/lock/ZookeeperLockRegistry.java (hunk 2 above)
Comment (artembilan) [reply spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/lock/ZookeeperLockRegistry.java]: Can you push these fixes by yourself to those branches?
Or do you want me to extract them somehow from such a big change? 😄 

Thanks

Comment (garyrussell) [reply spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/lock/ZookeeperLockRegistry.java]: No; I'll do it after review; I just didn't want to forget.

**Pull request merged**